### PR TITLE
fam_copy data items across memory server

### PR DESCRIPTION
**Pull request by faizan-barmawer**
fam_copy between data item from different memory server to be performed from destination memory server.

### CODE REVIEW

**Comment by faizan-barmawer** [comment]
Updated the PR with all the changes.
- Removed commented out code.
- added all the changes required from memory server side.

File: src/fam-api/fam_ops_libfabric.cpp [hunk 1]
```diff
@@ -177,6 +180,10 @@ int Fam_Ops_Libfabric::initialize() {
                                 message.str().c_str());
             }
 
+            // Save memery server address in memServerAddrs map
```
Comment (mashoodk) [inline src/fam-api/fam_ops_libfabric.cpp:183]:
```suggestion
            // Save memory server address in memServerAddrs map
```

**Review by mashoodk — CHANGES_REQUESTED** [review]
very minor comment.

**Comment by faizan-barmawer** [comment]
Pushed a commit with additional changes:
    - Added a pthread_rw_lock around FiMemsrvMap, required on server side, since memory-server is a multi-threaded application
    - Moved FiMemsrvMap from memory_registrion_libfabric.h to fam_ops_libfabric
    - Added a test case

Regards,
Faizan

**Review by mashoodk — APPROVED** [review]

**Pull request merged**